### ISSUE TITLE
Could `Optional.orElse(T)` be annotated with `PolyNull`

### ISSUE THREAD

**Issue by authchir** [typetools/checker-framework#1000]
Hello,
The method [`Optional.orElse(T)`][1] is currently tagged as follow.

     public @Nullable T orElse(@Nullable T other)

Which means the following code does not type check:

```Java
import java.util.Optional;

class Test {
  String foo() {
    Optional<String> opt = Optional.of("abc");
    return opt.orElse("");
  }
}
```
And instead, give the following error:
```
$ javacheck -version
javac 1.8.0-jsr308-2.1.6
$ javacheck -processor NullnessChecker Test.java
Test.java:6: error: [return.type.incompatible] incompatible types in return.
    return opt.orElse("");
                     ^
  found   : @Initialized @Nullable String
  required: @Initialized @NonNull String
1 error
```

Would not it make sense to change the annotation to use [`PolyNull`][2] instead?

[1]: https://github.com/typetools/checker-framework/blob/checker-framework-2.1.6/checker/jdk/nullness/src/java/util/Optional.java#L253
[2]: http://types.cs.washington.edu/checker-framework/current/api/org/checkerframework/checker/nullness/qual/PolyNull.html

**Comment by mernst** [comment]
You are right.

Your report led me to another problem with the `Optional` annotated JDK.  The type argument must be a non-null type; for example, the instantiation `Optional<@Nullable String>` should be illegal.

I'll fix both issues by updating the annotated JDK.
